flocker/node/agents/blockdevice.py

```diff
@@ -946,8 +1220,11 @@ def calculate_changes(self, configuration, cluster_state):
         )
 
```

Comment:
Resizes, deletes, etc. should only happen if no application is using the dataset; see logic in the P2PManifestationDeployer. File followup issue.